uaclient/entitlements/tests/test_base.py

```diff
@@ -371,24 +311,71 @@ def test_can_enable_when_incompatible_service_found(
             with mock.patch.object(
                 ent, "ENTITLEMENT_CLASS_BY_NAME", {"test": m_entitlement_cls}
             ):
-                ret, reason = base_ent.can_enable(allow_disable=allow_disable)
```

Comment:
I know this is a `can_enable` test, but since we moved the logic of handling incompatible services to enable, we should have an test there that is similar to this ones, were we verify that enable doesn't fail if allow disabling the incompatible service

Reply:
@lucasmoura The following test does ensure that `handle_incompatible_services` is called when `can_enable` returns `INCOMPATIBLE_SERVICE`. It hard-codes `handle_incompatible_services` to return `False` though.

Do you think we should add more? We could add a test where it calls `handle_incompatible_services`, which returns `True`, and then we verify that it proceeds to call `_perform_enable`?

Reply:
I think we could add one directly tied to `enable`, where we call enable with an incompatible service on and see if everything is happening according to plan

Reply:
Good idea I can add that

Reply:
@lucasmoura I added `test_enable_when_incompatible_service_found` which is basically the same as the old `test_can_enable_when_incompatible_service_found` but calls `enable` instead of `can_enable`